src/m5py/main.py

```diff
@@ -210,8 +211,14 @@ def fit(self, X, y: np.ndarray, sample_weight=None, check_input=True, X_idx_sort
         if self.use_smoothing not in [False, np.bool_(False), "installed", "on_prediction"]:
             raise ValueError("use_smoothing: Unexpected value: %s, please report it as issue." % self.use_smoothing)
 
+        # Get the correct fit method name based on the sklearn version used
+        sklearn_version_tuple = tuple(map(int, sklearn_version.split('.')))
+        fit_method_name = "fit" if sklearn_version_tuple <= (1, 3) else "_fit"
+
         # (1) Build the initial tree as usual
-        super(M5Base, self).fit(X, y, sample_weight=sample_weight, check_input=check_input)
+        fit_method = getattr(super(M5Base, self), fit_method_name)
```

Comment:
Nitpick: readability: please move all new lines after the comment, or all lines before the comment.